### DISCUSSION TITLE
so-dockerregistry  Missing/Pending

### DISCUSSION THREAD

**Discussion by Git-Me-Some-Hub** [Security-Onion-Solutions/securityonion#4901] [Unsupported Versions] [answered]
I noticed that my  container is has a status of missing or pending, but never green. 

I am now getting errors when trying to update via soup. 
`error parsing HTTP 404 response body: no error details found in HTTP response body: "{\"message\": \"The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.\"}\n"`

I saw something in the documentation about this command: `salt-call state.highstate` , but get this
`[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 500: driver failed programming external connectivity on endpoint so-dockerregistry (5169c371b5c5d0afc972041c3853d5792395470117e912d1e7569df514e13fc0): Error starting userland proxy: listen tcp4 0.0.0.0:5000: bind: address already in use'.`

I do see that port in use: 
![image](https://user-images.githubusercontent.com/17930434/126712520-bb2c3db1-516b-4e30-b504-4840f77bae33.png)

This ended with successful: 640 failed: 1

Seeing this still. No Filebeat director at all: `Exiting: error loading config file: stat /usr/share/filebeat/module-setup.yml: no such file or directory`

**Comment by TOoSmOotH** [comment]
You can try: sudo docker stop so-dockerregistry 

Then run soup again.

**Reply by Git-Me-Some-Hub** [reply]
Sadly the same issues is present. 

Here is it with soup being run: 
![image](https://user-images.githubusercontent.com/17930434/126927393-c4177566-5352-4b85-9cca-bfcb4979c0fd.png)

Here is with trying to start the docker container: 
![image](https://user-images.githubusercontent.com/17930434/126927143-247161b6-3ceb-400c-a171-7075ac0e6d1d.png)

Still seeing Python using that port: (this is after the docker stop command) 
![image](https://user-images.githubusercontent.com/17930434/126927959-21641ad6-d605-4f29-9117-27c78c139018.png)

Not sure if removing the container totally would help or is SO has the ability to proper rebuild it, if that is even something that could help. I ran a docker ps, and see all the images using port "server:5000/<image>"
![image](https://user-images.githubusercontent.com/17930434/126928808-eafee67a-0821-40ea-8b4d-0ecf969512c4.png)

Hope this helps.

**Reply by TOoSmOotH** [reply]
Is this a standalone install? If so it is probably strelka taking the port. I would try rebooting if it is a standalone.

**Reply by Git-Me-Some-Hub** [reply]
Sorry I forgot to include that part. This is a distributed install from the ISO, and this node is a managersearch. I have tried rebooting a couple times to no avail. 
soup fails if the so-status is pending of missing.

**Reply by Git-Me-Some-Hub** [reply]
I am seeing the error talk about userland proxy, is there any way to change this around to a hairpin proxy and have it start correctly?

looking at docker daemon.json I see this is using port 500:
![image](https://user-images.githubusercontent.com/17930434/127029647-31fb7f48-31b8-4d9c-aba9-aa4c20990d3e.png)


Anyway to completely kill the image and have it download a new one correctly?

**Reply by Git-Me-Some-Hub** [reply]
Any possible update? Maybe direction I can start looking?

**Comment by Git-Me-Some-Hub** [comment, marked as answer]
I will break out of the other thread to provide the answer to this. 

My Server cluster is virtual sitting on a Nutanix Cluster, and I thought it would be a good idea to have the Agent Tools installed. Welp! Apparently the client tools sit on port 5000, which is exactly the same as so-dockerregistry.... Meaning no updates via soup. I did not notice it right away, as it had port 5000 held, but must have been taken over by Nutanix after a reboot. 

It was listing python3 as the process, which SO uses, but when I grep for it I did not see it listed with all the others. 
`ps aux | grep python3`

So I went looking just for the PID listed in netstat 
`netstat -pant | grep 5000`
`ps aux | grep 1077`

That is where is gave me the path to Nutanix. Tried to stop it, disable it, but the only thing that helped clear up the port was uninstalling it. 
`/usr/local/nutanix/ngt/python36/python3 /usr/local/nutanix/ngt/python/bin/uninstall_ngt.py`

After that I started the docker per Mike's recommendation: 
`sudo docker stop so-dockerregistry`

all is well, and soup is running!!
